gc/base/standard/ConcurrentOverflow.cpp

```diff
@@ -114,7 +114,7 @@ MM_ConcurrentOverflow::emptyToOverflow(MM_EnvironmentBase *env, MM_Packet *packe
 
 	/* Broadcast the overflow to the concurrent collector
 	 * so it can take any remedial action */
-	collector->concurrentWorkStackOverflow();
+	collector->workStackOverflow();
```

Comment:
Chances are, we could move clearCardsForNewSpace() call to workStackOverflow(), in which case whole this method (and overflowItem()) would not need specialization for Concurrent. 
Ahtough, I'm not completely sure if the order of invocations is relevant (it would be slightly changed by the suggested grouping).

Reply:
This looks doable, from what I've found the ordering doesn't matter

Reply:
Only issue here is moving the implementation up to parent class isn't straight forward.

`ConcurrentOverflow` and `OverflowStandard` have the same parent `WorkPacketOverflow`, which currently has these methods as `Assert_MM_unreachable`. Since by design this wasn't meant to have an implementation, I'm not sure if we want to move it up given that all overflowhandlers inherit from this one class.

Reply:
- I suppose it doesn't matter moving the implementation up to this high level class since all other subclasses will implement their own.

or

- Another possibility is to have ConcurrentOverflow inherit from OverflowStandard instead
  - Will need to look into this and I would prefer to do this in follow up PR

Reply:
>If you think ConcurrentOverflow could be made a subclass of OverflowStandard without causing new problems, then we could try all this.

Would prefer to leave it as is for now and come back or do a follow up PR to make this change... shortly after this is merged